### PR TITLE
Update to using Model Optimizer (formerly AMMO) in PTQ workflow

### PR DESCRIPTION
**Pull request by janekl**
# What does this PR do ?

Summary:
* use Model Optimizer library (formerly AMMO) for LLM PTQ workflow
* restore PTQ tests failing due to setting `apply_rope_fusion = True` by default
* update documentation

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [x] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/export/quantize/quantizer.py
Comment (kevalmorabia97) [inline nemo/export/quantize/quantizer.py:205]: Please note that `def forward_loop(model): ...` is recommended over `def forward_loop(): ...` and the latter one will be deprecated soon

**Review by pablo-garay — APPROVED** [review]

File: nemo/export/quantize/quantizer.py
Comment (janekl) [reply nemo/export/quantize/quantizer.py:236]: changed

File: examples/nlp/language_modeling/megatron_quantization.py
Comment (janekl) [reply examples/nlp/language_modeling/megatron_quantization.py:1]: Maybe.. WDYT @ericharper ?

Comment (janekl) [reply examples/nlp/language_modeling/megatron_quantization.py:1]: Background is that the script is supposed to support multiple models -- gpt, llama, gemma etc. by specifying [decoder_type](https://github.com/NVIDIA/NeMo/blob/jlasek/ammo2modelopt/examples/nlp/language_modeling/conf/megatron_quantization.yaml#L31) param -- so I decided to rename it megatron_llama_quantization.py -> megatron_quantization.py.

**Comment by pablo-garay** [comment]
Can we look to merge this PR soon? @janekl @kevalmorabia97

**Comment by kevalmorabia97** [comment]
@pablo-garay no objection from my side. I don't have permissions to approve but this LGTM

File: examples/nlp/language_modeling/megatron_quantization.py
Comment (janekl) [reply examples/nlp/language_modeling/megatron_quantization.py:1]: Let's postpone final decision after merge as this name also is tied to the one used in [NeMo-Launcher](https://github.com/NVIDIA/NeMo-Framework-Launcher/blob/8b3527ae2809fb31933ce390a4ccbd262534aa7f/launcher_scripts/nemo_launcher/core/stages.py#L2461)

**Review by oyilmaz-nvidia — APPROVED** [review]
LGTM.

**Pull request merged**
